https://hg.mozilla.org/mozilla-central/rev/f7e2cf808d30

ld: warning: undefined symbol: wl_proxy_marshal_flags
>>> referenced by Unified_cpp_gfx_gl0.cpp
>>>               objdir/gfx/gl/Unified_cpp_gfx_gl0.o:(mozilla::gl::DeleteWaylandGLSurface(void*))
>>> referenced by Unified_cpp_gfx_gl0.cpp
>>>               objdir/gfx/gl/Unified_cpp_gfx_gl0.o:(mozilla::gl::GLContextEGL::CreateWaylandBufferSurface(mozilla::gl::EglDisplay&, void*, mozilla::gfx::IntSizeTyped<mozilla::gfx::UnknownUnits>&))
>>> referenced by Unified_cpp_gfx_layers1.cpp
>>>               objdir/gfx/layers/Unified_cpp_gfx_layers1.o:(mozilla::layers::NativeLayerRootWayland::CommitToScreen(mozilla::detail::BaseAutoLock<mozilla::Mutex&> const&))
>>> referenced 84 more times

diff --git widget/gtk/mozwayland/mozwayland.c widget/gtk/mozwayland/mozwayland.c
index 7a448e6321e2..1a86468b4f3d 100644
--- widget/gtk/mozwayland/mozwayland.c
+++ widget/gtk/mozwayland/mozwayland.c
@@ -200,3 +200,10 @@ MOZ_EXPORT int wl_list_empty(const struct wl_list* list) { return -1; }
 
 MOZ_EXPORT void wl_list_insert_list(struct wl_list* list,
                                     struct wl_list* other) {}
+
+MOZ_EXPORT struct wl_proxy* wl_proxy_marshal_flags(
+    struct wl_proxy* proxy, uint32_t opcode,
+    const struct wl_interface* interface, uint32_t version, uint32_t flags,
+    ...) {
+  return NULL;
+}
